### DISCUSSION TITLE
Where to keep / document our extensions to the font-format?

### DISCUSSION THREAD

**Discussion by behdad** [harfbuzz/harfbuzz#2896] [Ideas] [answered]
I'm going to start implementing things in HB that are not in any posted specification. Namely, to start with proposals Martin & I produced in 2016, which are currently in:

https://github.com/OpenType/opentype-layout

Obviously, that github org & repo name proved not accurate. So I'm asking for ideas and convergence on:

1. Where to make proposals to font-format changes to what HarfBuzz understands,
2. Where to document these, such that other tools & engineers / designers can read.

Another way to look at it is that we need our own font format website ala Apple's vs MS's etc.

@simoncozens Is CommonType something we can embrace for this?

Ideally whatever we choose will also explain how other engines differ, so CommonType might indeed be what we need and HB will become just one of the engines it covers. I've said this before as well: I wish we had the Mozilla Developer Network equivalent for font formats.

Please discuss but let's reach conclusion.

Some ideas for the documentation target, if a separate website sound not feasible, would be in our existing documentation structure. That sounds easiest. Another is the HB github wiki. I'm not very fond of that since I prefer to keep the documentation in sync with code.

cc @mhosken @jfkthame @anthrotype @moyogo @justvanrossum @madig @khaledhosny @adrientetar @rsheeter @davelab6 @m4rc1e @letterror @dscorbett @schriftgestalt @twardoch @alerque @bungeman

**Comment by simoncozens** [comment]
CommonType would work for this. There’s supposed to be a “CommonType Next” track for CT2.0 ideas. I keep meaning to sketch out an idea for a LAYT table - will try to get something written down soon.

**Comment by khaledhosny** [comment, marked as answer]
I think HarfBuzz documentation would be the natural place to document HarfBuzz-specific extensions, and the simplest to keep in sync with the code.
